src/components/SideNavigation/SideNavigationBase/SideNavigationBase.tsx

```diff
@@ -35,10 +39,11 @@ const SideNavigationBase = <C,>({
   icon,
   label,
   status,
+  forwardRef,
   ...props
 }: Props<C>) => {
   return (
-    <Component {...props}>
+    <Component ref={forwardRef} {...props}>
```

Comment:
Yes that's what I suspect the problem is. Some components can not receive a ref. Instead of restricting Component type to only have components that can receive a ref, since forwardRef is an optional prop is it possible to check if it's defined then only use this restriction?

Reply:
I don't think you can restrict a type dependency like that easily. Probably also not necessary. Why not restrict the component prop to a suitable type?

Reply:
Any idea how to restrict the component prop to guarantee that it can take a ref in?

Here's what I've tried but it doesn't seem to work:
```js
    component: ElementType<C> | ComponentType<C & { ref: React.Ref<C> }>;
```
I get the following error on the usage of SideNavigationBase:
```
Type '{ className: string; component: "a" | Props<L>["component"]; forwardRef: ((((instance: L) => void) | RefObject<L>) & ((instance: E) => void)) | ((((instance: L) => void) | RefObject<...>) & RefObject<...>); } & Omit<...>' is not assignable to type 'IntrinsicAttributes & { component: ElementType<L> | ComponentType<L & { ref: Ref<L>; }>; dark?: boolean; icon?: string; label: ReactNode; status?: ReactNode; forwardRef?: Ref<...>; } & Omit<...>'.
  Type '{ className: string; component: "a" | Props<L>["component"]; forwardRef: ((((instance: L) => void) | RefObject<L>) & ((instance: E) => void)) | ((((instance: L) => void) | RefObject<...>) & RefObject<...>); } & Omit<...>' is not assignable to type 'Omit<L, "component" | "dark" | "icon" | "label" | "status" | "forwardRef">'.
  ```

Reply:
The problem lies in SideNavigationLink, there you need
```
export type Props<L = LinkDefaultElement, E = HTMLAnchorElement> = Partial<
  Omit<SideNavigationBaseProps<L>, "component">
> & {
  /**
   * The navigation item's label.
   */
  label: ReactNode;
```
in the lines 10ff.

Reply:
Thanks David, that did indeed solved the issue 🙏